core-processor/src/main/java/io/micronaut/inject/processing/DeclaredBeanElementCreator.java

```diff
@@ -324,17 +331,17 @@ protected boolean visitPropertyWriteElement(BeanDefinitionVisitor visitor,
      * @return true if processed
      */
     protected boolean visitMethod(BeanDefinitionVisitor visitor, MethodElement methodElement) {
-        if (visitInjectAndLifecycleMethod(visitor, methodElement)) {
+        if (visitInjectAndLifecycleMethod(visitor, null, methodElement)) {
             return true;
         }
         return visitAopAndExecutableMethod(visitor, methodElement);
     }
 
     @NextMajorVersion("Require @ReflectiveAccess for private methods in Micronaut 4")
-    private boolean visitInjectAndLifecycleMethod(BeanDefinitionVisitor visitor, MethodElement methodElement) {
+    private boolean visitInjectAndLifecycleMethod(BeanDefinitionVisitor visitor, @Nullable PropertyElement propertyElement, MethodElement methodElement) {
         // All the cases above are using executable methods
         boolean claimed = false;
-        if (methodElement.hasDeclaredAnnotation(AnnotationUtil.POST_CONSTRUCT)) {
```

Comment:
Here also property's annotation metadata should be properly propagated to the method so you don't need to use property at all

Reply:
does this refer to the correct line of code? *confused*